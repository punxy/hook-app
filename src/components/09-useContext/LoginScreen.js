import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const user = {
        id: 123,
        name: 'Sebasti√°n Silva',
        email: 'sebasilvac88@gmail.com'
    }

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />

            <button
                className='btn btn-primary'
                onClick={() => { setUser(user) }}
            >setUser</button>
        </div>
    )
}
