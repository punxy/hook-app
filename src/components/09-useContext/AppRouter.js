import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
} from "react-router-dom";
import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className='container-fluid'>
                    {/* Version 6 */}
                    <Routes>
                        <Route path="/" element={<HomeScreen />} />
                        <Route path="login" element={<LoginScreen />} />
                        <Route path="about" element={<AboutScreen />} />
                        <Route path='*' element={<HomeScreen />} />
                    </Routes>
                    {/* Version 5
                        <Switch>
                        <Route exact path='/' component={ HomeScreen }/>
                        <Route exact path='/about' component={ AboutScreen }/>
                        <Route exact path='/login' component={ LoginScreen }/>
                    </Switch> */}
                </div>
            </div>
        </Router>
    )
}
