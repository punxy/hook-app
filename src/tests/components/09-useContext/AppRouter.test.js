import React from 'react';
import { mount } from "enzyme";
import '@testing-library/jest-dom'; 
import { UserContext } from '../../../components/09-useContext/UserContext';
import { AppRouter } from '../../../components/09-useContext/AppRouter';

describe('Test in AppRouter component', () => {

    const user = {
        name: 'Sebasti√°n Silva',
        email: 'sebasilvac88@gmail.com'
    };

    const wrapper = mount(
        <UserContext.Provider value={{user}} >
            <AppRouter />
        </UserContext.Provider>
    );

    test('should be render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});