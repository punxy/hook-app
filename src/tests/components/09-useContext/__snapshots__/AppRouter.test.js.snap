// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AppRouter component should be render correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          activenavlink="active"
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          activenavlink="active"
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          activenavlink="active"
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container-fluid"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "Sebasti√°n Silva",
   "email": "sebasilvac88@gmail.com"
}
      </pre>
    </div>
  </div>
</div>
`;
